gazelle_haskell_modules/lang.go

```diff
@@ -180,7 +180,14 @@ func (*gazelleHaskellModulesLang) Resolve(c *config.Config, ix *resolve.RuleInde
 		gazelle_cabal.RunResolve(c, ix, rc, r, expr, from)
 		// We then create an HRuleImportData from the generated rule.
 		newr, newImports := addOneNonHaskellModuleRule(c.RepoRoot, from.Repo, from.Pkg, r)
-		setNonHaskellModuleDeps(&hmc, c.RepoRoot, ix, newr, newImports, from)
+		// The new version of the rule is used to clean the attributes of r.
+		for _, a := range r.AttrKeys() {
+			if newr.Attr(a) == nil {
+				r.DelAttr(a)
+			}
+		}
+		// After this cleaning, the new deps are set.
+		setNonHaskellModuleDeps(&hmc, c.RepoRoot, ix, r, newImports, from)
```

Comment:
Why are the attributes deleted from the old rule?

Reply:
Sorry for the very delayed answer.
Those attributes are deleted because `Resolve` is not a pure function, it alters the rules given to it. What I really want to do is to replace `r` by `newr`, but this not possible (in the sense that I do whatever is required on `newr`, this will not impact `r` and in the end, `newr` won't exist outside of the scope of this function). So I have to copy `newr` into `r` and for this, I use my knowledge that the difference between the 2 is the existence of some attributes to do it by deletion rather than doing a "deep copy".

Reply:
Definitely worth adding a comment.